src/app/shared/components/openstack-credentials/application/component.ts

```diff
@@ -99,9 +103,26 @@ export class OpenstackProviderBasicAppCredentialsComponent extends BaseFormValid
     this.form = this._builder.group({
       [Controls.ApplicationCredentialID]: this._builder.control('', Validators.required),
       [Controls.ApplicationCredentialSecret]: this._builder.control('', Validators.required),
-      [Controls.FloatingIPPool]: this._builder.control('', Validators.required),
+      [Controls.FloatingIPPool]: this._builder.control(''),
     });
 
+    merge(
+      this.form.get(Controls.ApplicationCredentialID).valueChanges.pipe(distinctUntilChanged()),
+      this.form.get(Controls.ApplicationCredentialSecret).valueChanges.pipe(distinctUntilChanged())
+    )
+      .pipe(debounceTime(this._debounceTime))
+      .pipe(takeUntil(this._unsubscribe))
+      .subscribe(_ => (this._clusterSpecService.cluster = this._getClusterEntity()));
+
+    // Look-up for presets and floating IP pool is only required when we want to populate fields based on either the
+    // enabled preset or the provided credentials
+    if (!this.verifyCredentials) {
+      this.form.reset();
+      return;
```

Comment:
This one will stop init method from running further.

Reply:
Yes, I don't want the next parts to be executed.

Reply:
Then revert the logic please as it's not clear enough by looking at the code.

Reply:
How is it not clear? I'm explicitly using `return` to exit.

Reply:
What I meant is putting the code that is below inside the if statement and reversing the condition. It's easier to notice that it will run only if some conditions will be meet.

Reply:
In general, we try to avoid **huge** IF blocks if we can return early. Also `Then revert the logic please as it's not clear enough by looking at the code.` **reverting** the logic doesn't imply **reversing** it. 

Anyhow, unfortunately, I won't be continuing with this PR. It's best for you to take the direction that suits you ✌️

Reply:
This is a good pattern when it comes to i.e. validation, but for me it's not best in discussed context. A new method would be better. It's rather small detail anyways. 

And yeah, my bad with using wrong word before.